src/server/scripts/KulTiras/ShrineOfTheStorm/boss_tidesage_council.cpp

```diff
@@ -0,0 +1,541 @@
+    SPELL_BLOWBACK_DAMAGE                      = 267838,
+    SPELL_BLOWBACK_AREATRIGGER                 = 267836,
+    SPELL_SWIFTNESS_WARD_SELECTOR              = 267902,
+    SPELL_SWIFTNESS_WARD_AREATRIGGER           = 267891,
+    SPELL_SWIFTNESS_WARD_HASTE_BOSS            = 267890,
+    SPELL_SWIFTNESS_WARD_HASTE_PLAYER          = 267888,  // Pets can get buff too
+    SPELL_COSMETIC_SWIFTNESS_WARD              = 274689,
+    SPELL_GALECALLER_FAYE_GAIN_ENERGY          = 267894 // Serverside
+};
+
+enum TidesageCouncilEvents
+{
+    EVENT_BLESSING_OF_IRONSIDES = 1,
+    EVENT_HINDERING_CLEAVE,
+    EVENT_REINFORCING_WARD,
+
+    EVENT_BLESSING_OF_THE_TEMPEST,
+    EVENT_SLICING_BLAST,
+    EVENT_SWIFTNESS_WARD,
+
+    EVENT_CHECK_MANA
+};
+
+enum TidesageCouncilTexts
+{
+    SAY_AGGRO                    = 0,
+    SAY_SLAY                     = 3,
+    SAY_WIPE                     = 4,
+    SAY_DEAD                     = 5,
+
+    // Galecaller Faye
+    SAY_SWIFTING_WARD_WARNING    = 1,
+    SAY_SWIFTING_WARD            = 2,
+    SAY_BLESSING_OF_THE_TEMPEST  = 6,
+    SAY_BROTHER_DIES             = 7,
+
+    // Brother Ironhull
+    SAY_REINFORCING_WARD         = 1,
+    SAY_REINFORCING_WARD_WARNING = 2,
+    SAY_FAYE_DIES                = 6
+};
+
+uint32 const TidesageData[2] =
+{
+    DATA_BROTHER_IRONHULL,
+    DATA_GALECALLER_FAYE
+};
+
+namespace
+{
+void DespawnTidesageCouncil(InstanceScript* instance, EvadeReason why)
+{
+    if (instance->GetBossState(DATA_TIDESAGE_COUNCIL) == FAIL)
+        return;
+
+    instance->SetBossState(DATA_TIDESAGE_COUNCIL, FAIL);
+
+    for (uint32 bossesData : TidesageData)
+    {
+        if (Creature* council = instance->GetCreature(bossesData))
+        {
+            instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, council);
+            council->AI()->EnterEvadeMode(why);
+        }
+    }
+}
+
+void TidesageCouncilEncounterDone(InstanceScript* instance)
+{
+    if (instance->GetBossState(DATA_TIDESAGE_COUNCIL) == DONE)
+        return;
+
+    for (uint32 bossesData : TidesageData)
+    {
+        if (Creature* council = instance->GetCreature(bossesData))
+        {
+            if (council->IsAlive())
+                return;
+        }
+    }
+
+    instance->SetBossState(DATA_TIDESAGE_COUNCIL, DONE);
+}
+}
+
+struct TidesageCouncilSharedAI : public BossAI
+{
+    TidesageCouncilSharedAI(Creature* creature, uint32 bossId) : BossAI(creature, bossId)
+    {
+        SetBoundary(instance->GetBossBoundary(DATA_TIDESAGE_COUNCIL));
+    }
+
+    void EnterEvadeMode(EvadeReason why) override
+    {
+        DespawnTidesageCouncil(instance, why);
+
+        Talk(SAY_WIPE);
+
+        events.Reset();
+        _EnterEvadeMode();
+        _DespawnAtEvade();
+    }
+
+    void JustEngagedWith(Unit* /*who*/) override
+    {
+        for (uint32 bossesData : TidesageData)
+        {
+            if (Creature* council = instance->GetCreature(bossesData))
```

Comment:
I'd check if council is alive and not in combat, neither of the functions inside it prevent DoZoneInCombat from happening